core/client/fs/src/main/java/alluxio/util/GrpcDefaultOptions.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util;
+
+import alluxio.client.ReadType;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.grpc.CheckConsistencyPOptions;
+import alluxio.grpc.CreateDirectoryPOptions;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.grpc.DeletePOptions;
+import alluxio.grpc.ExistsPOptions;
+import alluxio.grpc.FileSystemMasterCommonPOptions;
+import alluxio.grpc.FreePOptions;
+import alluxio.grpc.GetStatusPOptions;
+import alluxio.grpc.ListStatusPOptions;
+import alluxio.grpc.LoadMetadataPType;
+import alluxio.grpc.MountPOptions;
+import alluxio.grpc.OpenFilePOptions;
+import alluxio.grpc.RenamePOptions;
+import alluxio.grpc.ScheduleAsyncPersistencePOptions;
+import alluxio.grpc.SetAclPOptions;
+import alluxio.grpc.SetAttributePOptions;
+import alluxio.grpc.TtlAction;
+import alluxio.grpc.UnmountPOptions;
+import alluxio.grpc.WritePType;
+
+/**
+ * This class contains static methods which can be passed Alluxio configuration objects that
+ * will populate the gRPC options objects with the proper values based on the given configuration.
+ */
+public class GrpcDefaultOptions {
```

Comment:
We have a master counterpart of this class named FileSystemMasterOptions and it's a util class.
For the sake of consistency can we rename it as FileSystemClientOptions? We had this intermediary class before and it was named as such. This logic is not gRPC related so we don't need to name it as such.

Reply:
I can agree with that. However I don't necessarily see the logic here as client-side only issues. After taking a look at `FileSystemMasterOptions` we have much of the same code. It's just that the class relies on `ServerConfiguration` which is unavailable to the client.

I see this functionality as something that's shared on both on the client and server side. The logic between both classes is effectively the same. Originally I was going to put `GrpcDefaultOptions` in `core-common` but it relied on the `core-transport` dependency which I didn't want to add, so I moved it into the client. If we do want to add that dependency and move the logic to `core-common` then both client and server can rely on the same logic to populate the options objects. The server side will just need to pass `ServerConfiguration.global()` to all of the methods.

Reply:
My motivation is to restrict this change's impact to client side. Server should have the authority to enforce defaults, and client should only send when it wants to override those defaults.